### PR TITLE
upstream: fix oss-fuzz issue #11095.

### PR DESCRIPTION
**Pull request by ipuustin**
Description:

Fix a crash found by oss-fuzz (see https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=11095). The bug is caused by trying to request IP information from a unix domain socket address.

Risk Level: low
Testing: local fuzzing and a regression test for `HostDescriptionImpl` constructor

### CODE REVIEW

**Comment by ipuustin** [comment]
Makes sense. I added a patch which does that.

File: source/common/upstream/upstream_impl.h [hunk 1]
```diff
@@ -69,17 +69,25 @@ class HostDescriptionImpl : virtual public HostDescription {
...
         locality_(locality), stats_{ALL_HOST_STATS(POOL_COUNTER(stats_store_),
                                                    POOL_GAUGE(stats_store_))},
-        priority_(priority) {}
+        priority_(priority) {
+    if (health_check_config.port_value() != 0 &&
+        dest_address->type() != Network::Address::Type::Ip) {
+      // Setting the health check port to non-0 only works for IP-type addresses. Setting the port
+      // for a pipe address is a misconfiguration. Throw an exception.
+      throw EnvoyException(
+          fmt::format("Invalid host configuration: non-null port for non-IP address"));
```
Comment (htuch) [inline source/common/upstream/upstream_impl.h]: Nit: s/non-null/non-zero/

**Comment by htuch** [comment]
@ipuustin LGTM, but needs DCO fix https://github.com/envoyproxy/envoy/blob/master/CONTRIBUTING.md#fixing-dco

**Comment by ipuustin** [comment]
@htuch Sure, I can help with them. There's also still a bunch of already open ones in the fuzzer backlog.

**Pull request merged**